src/Analyzers/Core/Analyzers/DiagnosticCustomTags.cs

```diff
@@ -33,7 +35,7 @@ public static string[] EditAndContinue
         {
             get
             {
-                Assert(s_editAndContinueCustomTags, WellKnownDiagnosticTags.EditAndContinue, WellKnownDiagnosticTags.Telemetry, WellKnownDiagnosticTags.NotConfigurable);
+                Assert(s_editAndContinueCustomTags, WellKnownDiagnosticTags.EditAndContinue, WellKnownDiagnosticTags.Telemetry, WellKnownDiagnosticTags.NotConfigurable, s_enforceOnBuildNeverTag);
```

Comment:
❔ Is there any value to providing `EnforceOnBuild` when `NotConfigurable` is also provided? If not, it would make it easier for me to understand the impact if the `EnforceOnBuild` tag was omitted from cases where `NotConfigurable` was used.

Otherwise, I would suggest adding a new `EnforceOnBuild.Unspecified` value to use in these cases.

Reply:
The intent was to keep `EnforceOnBuild` completely independent concept, regardless of underlying tags. Under the covers, we ensure that diagnostics marked as `NotConfigurable` tag always pass in `EnforceOnBuild.Never` with an assert in `DiagnosticCustomTags.Create method`, but reverse is not always true (for example IDE0079 is an IDE-only analyzer that cannot be enforced on build, but can be configured in the IDE to a different severity.